src/PowerShell/CommonFiles/PowerShellCmdlet.cs

```diff
@@ -603,6 +596,11 @@ private void WaitMainThreadActionCompletion()
                 this.pwshThreadActionCompleted.WaitHandle,
             });
 
+            if (this.pwshThreadActionException != null)
+            {
+                throw this.pwshThreadActionException;

```

Comment:
Should this throw a new exception with `pwshThreadActionException` as the inner exception?  That way we get to keep the original stack trace.